pkg/manifest/installer.go

```diff
@@ -77,6 +77,15 @@ var (
 	istioVersionLabelStr = name.OperatorAPINamespace + "/version"
 )
 
+// ApplyResult is typed to constrain allowed values to represent the result of kubectl apply.
+type ApplyResultType string
+
+const (
+	ApplyManifest ApplyResultType = "ApplyManifest"
+	EmptyManifest ApplyResultType = "EmptyManifest"
+	ParseManifest ApplyResultType = "ParseManifest"
+)
```

Comment:
This API is confusing. We are mixing an action (apply/parse) with an attribute (empty). It's also not clear why we would need to label a manifest as "empty". If an action was performed on an empty manifest and we need to know about that, let's create a relationship between the action and the manifest that is maintained to the point that we need to know whether the manifest was empty.

Reply:
Removed.